source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/OutboxMessageProcessorEndpointTests.cs

```diff
@@ -107,53 +141,100 @@ public async Task RunAsync_WhenOutboxMessageIsRead_AvailableDataIsPersisted_AndP
 
                 // Act & Assert
                 dispatcher.Setup(d => d.DispatchAsync(
-                        It.IsAny<ChargePriceOperationsRejectedEvent>(),
-                        It.IsAny<CancellationToken>()))
-                    .Throws<Exception>();
+                        It.IsAny<InternalEvent>(),
+                        It.IsAny<CancellationToken>())).Throws<Exception>();
 
                 await Assert.ThrowsAsync<Exception>(() => sut.RunAsync(timerInfo));
 
                 dispatcher.Setup(d => d.DispatchAsync(
-                        It.IsAny<ChargePriceOperationsRejectedEvent>(),
+                        It.IsAny<InternalEvent>(),
                         It.IsAny<CancellationToken>()))
-                    .Callback<ChargePriceOperationsRejectedEvent, CancellationToken>((_, _) => { });
+                    .Callback<InternalEvent, CancellationToken>((_, _) => { });
                 await sut.RunAsync(timerInfo);
 
                 outboxMessage = chargesDatabaseReadContext.OutboxMessages.Single(x => x.Id == outboxMessage.Id);
                 outboxMessage.ProcessedDate.Should().Be(now);
             }
 
-            private static ChargePriceOperationsRejectedEvent CreateChargePriceOperationsRejectedEvent()
+            [Fact]
+            public async Task GivenNewRejectedEvent_WhenRunAsync_ThenRejectedEventIsProcessed()
+            {
+                // Arrange
+                var messageType = typeof(PriceRejectedEvent).FullName!;
+                await using var chargesDatabaseWriteContext = _databaseManager.CreateDbContext();
+                await using var chargesDatabaseReadContext = _databaseManager.CreateDbContext();
+                var jsonSerializer = new JsonSerializer();
+                var rejectedEvent = new PriceRejectedEventBuilder().Build();
+                var type = messageType;
+                var serializedEvent = jsonSerializer.Serialize(rejectedEvent);
+                var outboxMessage = new OutboxMessageBuilder().WithType(type).WithData(serializedEvent).Build();
+                chargesDatabaseWriteContext.OutboxMessages.Add(outboxMessage);
+                await chargesDatabaseWriteContext.SaveChangesAsync();
+                outboxMessage.ProcessedDate.Should().BeNull();
+
+                // Act
+                await FunctionAsserts.AssertHasExecutedAsync(
+                    Fixture.HostManager, nameof(OutboxMessageProcessorEndpoint));
+
+                // Assert
+                var processedOutboxMessage = chargesDatabaseReadContext.OutboxMessages.Single(x => x.Id == outboxMessage.Id);
+                processedOutboxMessage.Should().BeEquivalentTo(outboxMessage, om => om.Excluding(p => p.ProcessedDate));
+                processedOutboxMessage.ProcessedDate.Should().NotBeNull();
+                processedOutboxMessage.Type.Should().Be(messageType);
+            }
+
+            private static PriceRejectedEvent CreateChargePriceOperationsRejectedEvent()
```

Comment:
Suggest we keep the names aligned

```suggestion
            private static PriceRejectedEvent CreatePriceRejectedEvent()
```